deploy/terraform/network.tf

```diff
@@ -0,0 +1,303 @@
+## VPC Module ##
+// All the below can be put in to a tf module setting up the vpc
+// outputs:
+// - public subnet ids
+// - private subnet ids
+
+resource "aws_vpc" "nia_gp2gp_vpc" {
+  cidr_block           = var.vpc_cidr_block
+  enable_dns_hostnames = true
+  enable_dns_support = true
+
+  tags = {
+    Name = "nia_gp2gp_vpc"
+  }
+}
+
+resource "aws_internet_gateway" "nia_gp2gp_igw" {
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+
+  tags = {
+    Name = "nia_gp2gp_igw"
+  }
+}
+
+resource "aws_subnet" "nia_gp2gp_public_subnet" {
+  count             = var.subnet_count.public
+  cidr_block        = var.public_subnet_cidr_blocks[count.index]
+  vpc_id            = aws_vpc.nia_gp2gp_vpc.id
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+
+  tags = {
+    Name = "nia_gp2gp_public_subnet_${count.index}"
+  }
+}
+
+resource "aws_subnet" "nia_gp2gp_private_subnet" {
+  count             = var.subnet_count.private
+  vpc_id            = aws_vpc.nia_gp2gp_vpc.id
+  cidr_block        = var.private_subnet_cidr_blocks[count.index]
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+
+  tags = {
+    Name = "nia_gp2gp_private_subnet${count.index}"
+  }
+}
+
+resource "aws_route_table" "nia_gp2gp_public_rt" {
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.nia_gp2gp_igw.id
+  }
+}
+
+resource "aws_route_table_association" "public" {
+  count          = var.subnet_count.public
+  route_table_id = aws_route_table.nia_gp2gp_public_rt.id
+  subnet_id      = aws_subnet.nia_gp2gp_public_subnet[count.index].id
+}
+
+resource "aws_route_table" "nia_gp2gp_private_rt" {
+  vpc_id = aws_vpc.nia_gp2gp_vpc.id
+}
+
+resource "aws_route_table_association" "private" {
+  count          = var.subnet_count.private
+  route_table_id = aws_route_table.nia_gp2gp_private_rt.id
+  subnet_id      = aws_subnet.nia_gp2gp_private_subnet[count.index].id
+}
+
+resource "aws_security_group" "nia_gp2gp_dmz" {
+  name        = "nia_gp2gp_api_sg"
+  description = "Public security group"
+  vpc_id      = aws_vpc.nia_gp2gp_vpc.id
+
+  ingress {
+    from_port   = var.http_server_port
+    protocol    = "tcp"
+    to_port     = var.http_server_port
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  //TODO: restrict access to match current restrictions
+
+  //ACCEPT TRAFFIC TO FACADE PORT 8081
+  ingress {
+    from_port = 8081
+    protocol  = "TCP"
+    to_port   = 8081
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Facade Traffic"
+  }
+
+  //ACCEPT TRAFFIC TO INBOUND PORT 443
+  ingress {
+    from_port = 443
+    protocol  = "TCP"
+    to_port   = 443
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Inbound Traffic"
+  }
+
+  //ACCEPT TRAFFIC TO MOCK SPINE PORT 8086
```

Comment:
Can you remind me how we use this mock spine?

Reply:
It's for testing purpose so we can use out Postman Test Suite to verify various things are working as they should. Wouldn't be useful in production.